### PR TITLE
fix(api): preserve typeless output schemas in tool conversion

### LINKED ISSUE #287: $defs/additionalProperties dropped from output schemas with no top-level type
`domainTool.ToAPIType` (`internal/api/tools.go:176`) only builds `outputSchema` when `d.OutputSchema.Type != ""`. An output schema that is valid JSON Schema but omits a top-level `type` — e.g. `{"$defs": {...}, "additionalProperties": false}` — is dropped entirely, including the `$defs`/`additionalProperties` fields threaded in by #279.

The `Type != ""` check is a proxy for "output schema present". It predates #279 and is only reachable on the output path (the input schema is always constructed).

Impact is low: MCP output schemas are spec'd as objects and almost always carry `type: "object"`, so a typeless-but-populated output schema is rare in practice.

Fix: gate `outputSchema` construction on whether any field is populated rather than on `Type` alone, and add a test covering a typeless output schema.

### PR DESCRIPTION
**Pull request by peteski22**
Fixes #287.

`domainTool.ToAPIType` gated output-schema construction on `d.OutputSchema.Type != ""`. A valid JSON Schema may omit a top-level `type` while still defining `$defs`/`additionalProperties`/`properties` (e.g. `{"$defs": {...}, "additionalProperties": false}`), so such a schema was dropped entirely — stranding any `$ref` whose `$defs` went with it.

## Change

- Add an `outputSchemaPresent` predicate that reports presence when any schema field is populated, not just when a top-level `type` is set. It treats `additionalProperties: false` as content and empty-but-non-nil collections as absence.
- Gate output-schema construction on the predicate.

## Tests

- `TestOutputSchemaPresent`: table-driven over the predicate's edge cases (each field alone, `additionalProperties: false`, empty-but-non-nil collections, zero value).
- `TestDomainTool_ToAPIType_OutputSchema`: table-driven over the conversion boundary — typeless preserved, empty dropped, typed preserved.

`make test` (lint + full suite) passes.

## Follow-up

A separate, pre-existing rendering quirk surfaced by this fix is tracked in #294: `JSONSchema.Type` has no `omitempty`, so a now-preserved typeless schema serializes with `"type": ""`.

## Summary by CodeRabbit

* **Bug Fixes**
  * Output schemas are now preserved when they include definitions, properties, required fields, or additional-property settings—even without a type declaration.
  * Schemas with `additionalProperties: false` are correctly retained.
  * Empty output schemas continue to be omitted, preventing unnecessary schema data from being exposed.

### CODE REVIEW

File: internal/api/tools.go [hunk 1]
```diff
@@ -173,7 +173,7 @@ func (d domainTool) ToAPIType() (Tool, error) {
 	}
 
 	var outputSchema *JSONSchema
-	if d.OutputSchema.Type != "" {
+	if outputSchemaPresent(d.OutputSchema) {
```
[the remark this answers is not in the document]
Comment (peteski22) [reply internal/api/tools.go:176]: See: https://github.com/mozilla-ai/mcpd/issues/294

**Review by HareeshBahuleyan — APPROVED** [review]
